app/sliders/plotter.py

```diff
@@ -1,202 +1,310 @@
+        )
+
+        # Add interactivity
+        self.filters_multi_choice.param.watch(self.update_filters, "value")
+        self.filters_single_choice.param.watch(self.update_filters, "value")
+
+    def update_filters(self, event):
+        """
+        Update data and plot whenever the user changes the filters.
+
+        :param event: an event object that triggers the update
+        """
+        for code in self.config["plot_codes"]:
+            # Re select data based on new selection of filters
+            selected_lines = self.filters_multi_choice.value
+            single_choice_dict = self.raw_data[code][self.filters_single_choice.value]
+
+            filtered_data = {
+                "x": single_choice_dict["x"],
+                **{line: single_choice_dict[line] for line in selected_lines}
+            }
 
-        # Add vertical bars to the figure using the ColumnDataSource
-        self.plot.vbar(x="x", top="y", source=self.data, **self.config["vbar"])
+            # Update the existing ColumnDataSource with new data
+            self.fitted_data[code].data = filtered_data
 
-        # Show the legend
-        self.plot.legend.title = self.config["legend_title"]
-        self.plot.legend.label_text_font_size = self.config["label_text_font_size"]
 
+class PiePlotter(Plotter):
+    def __init__(self, raw_data, config):
+        """
+        Initialize a plotter for creating pie charts.
+
+        :param raw_data: dict, data to plot
+        :param config: dict, default configuration
+        """
+        super().__init__(raw_data, config)
+
+    def fit_data(self):
+        """
+        Prepare data.
+        """
+        x_values = self.raw_data["x"]
+        y_values = self.raw_data["y"]
+
+        # Calculate area for each category in the pie chart
+        total = sum(y_values)
+        angles = [2 * pi * (y / total) for y in y_values]
+        colors = Category20c[len(x_values)]
+
+        # Transform data to the ColumDataSource format required by Bokeh
+        data = {"x": x_values,
+                "y": y_values,
+                "angle": angles,
+                "color": colors}
+        source = ColumnDataSource(data)
+
+        self.fitted_data = source
 
-class PlotterFactory:
-    @staticmethod
-    def create_plotter(plot_type):
-        class_name = plot_type.capitalize() + "Plotter"
-        cls = globals()[class_name]
-        return cls()
+    def create_plot(self):
+        """
+        Create plot.
+        """
+        # Create a Bokeh figure
+        plot = figure(**self.config["general"])
+
+        plot.wedge(**self.config["wedge"],
+                   source=self.fitted_data,
+                   start_angle=cumsum("angle", include_zero=True),
+                   end_angle=cumsum("angle"))
+        plot.axis.axis_label = self.config["axis_label"]
+        plot.axis.visible = self.config["visible"]
+        plot.grid.grid_line_color = self.config["grid_line_color"]
+
+        self.plot = plot
```

Comment:
just a personal preference, I think the plotter.py file is a bit packed and the chart specific classes, like BarPlotter(Plotter), might also get there own files.

Reply:
I'll think about it!